tests/components/netatmo/test_diagnostics.py

```diff
@@ -0,0 +1,94 @@
+"""Test the Netatmo diagnostics."""
+from unittest.mock import AsyncMock, patch
+
+from homeassistant.setup import async_setup_component
+
+from .common import fake_post_request
+
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+
+
+async def test_entry_diagnostics(hass, hass_client, config_entry):
+    """Test config entry diagnostics."""
+    with patch(
+        "homeassistant.components.netatmo.api.AsyncConfigEntryNetatmoAuth",
+    ) as mock_auth, patch(
+        "homeassistant.helpers.config_entry_oauth2_flow.async_get_config_entry_implementation",
+    ), patch(
+        "homeassistant.components.netatmo.webhook_generate_url"
+    ):
+        mock_auth.return_value.async_post_request.side_effect = fake_post_request
+        mock_auth.return_value.async_addwebhook.side_effect = AsyncMock()
+        mock_auth.return_value.async_dropwebhook.side_effect = AsyncMock()
+        assert await async_setup_component(hass, "netatmo", {})
+
+    await hass.async_block_till_done()
+
+    result = await get_diagnostics_for_config_entry(hass, hass_client, config_entry)
+
+    # ignore for tests
+    result["info"]["data"]["token"].pop("expires_at")
+    result["info"].pop("entry_id")
+
+    assert result["info"] == {
+        "data": {
+            "auth_implementation": "cloud",
+            "token": {
+                "access_token": "mock-access-token",
+                "expires_in": 60,
+                "refresh_token": "mock-refresh-token",
```

Comment:
Redact the access and refresh tokens.

Reply:
Oh, good point. Missed that due to the mocking. 😬